arm/Microsoft.ServiceFabric/clusters/deploy.bicep

```diff
@@ -0,0 +1,355 @@
+  tableEndpoint: !empty(diagnosticsStorageAccountConfig) ? diagnosticsStorageAccountConfig.tableEndpoint : null
+}
+
+var fabricSettings_var = [for fabricSetting in fabricSettings: {
+  name: !empty(fabricSetting.name) ? fabricSetting.name : null
+  parameters: !empty(fabricSetting.parameters) ? fabricSetting.parameters : null
+}]
+
+var nodeTypes_var = [for nodeType in nodeTypes: {
+  applicationPorts: contains(nodeType, 'applicationPorts') ? {
+    endPort: contains(nodeType.applicationPorts, 'endPort') ? nodeType.applicationPorts.endPort : null
+    startPort: contains(nodeType.applicationPorts, 'startPort') ? nodeType.applicationPorts.startPort : null
+  } : null
+  capacities: contains(nodeType, 'capacities') ? nodeType.capacities : null
+  clientConnectionEndpointPort: contains(nodeType, 'clientConnectionEndpointPort') ? nodeType.clientConnectionEndpointPort : null
+  durabilityLevel: contains(nodeType, 'durabilityLevel') ? nodeType.durabilityLevel : null
+  ephemeralPorts: contains(nodeType, 'ephemeralPorts') ? {
+    endPort: contains(nodeType.ephemeralPorts, 'endPort') ? nodeType.ephemeralPorts.endPort : null
+    startPort: contains(nodeType.ephemeralPorts, 'startPort') ? nodeType.ephemeralPorts.startPort : null
+  } : null
+  httpGatewayEndpointPort: contains(nodeType, 'httpGatewayEndpointPort') ? nodeType.httpGatewayEndpointPort : null
+  isPrimary: contains(nodeType, 'isPrimary') ? nodeType.isPrimary : null
+  isStateless: contains(nodeType, 'isStateless') ? nodeType.isStateless : null
+  multipleAvailabilityZones: contains(nodeType, 'multipleAvailabilityZones') ? nodeType.multipleAvailabilityZones : null
+  name: '${(!empty(nodeType.name) ? nodeType.name : 'Node00')}'
+  placementProperties: contains(nodeType, 'placementProperties') ? nodeType.placementProperties : null
+  reverseProxyEndpointPort: contains(nodeType, 'reverseProxyEndpointPort') ? nodeType.reverseProxyEndpointPort : null
+  vmInstanceCount: contains(nodeType, 'vmInstanceCount') ? nodeType.vmInstanceCount : 1
+}]
+
+var notifications_var = [for notification in notifications: {
+  isEnabled: contains(notification, 'isEnabled') ? notification.isEnabled : false
+  notificationCategory: !empty(notification.notificationCategory) ? notification.notificationCategory : 'WaveProgress'
+  notificationLevel: !empty(notification.notificationLevel) ? notification.notificationLevel : 'All'
+  notificationTargets: !empty(notification.notificationTargets) ? notification.notificationTargets : []
+}]
+
+var reverseProxyCertificate_var = {
+  thumbprint: !empty(reverseProxyCertificate) ? reverseProxyCertificate.thumbprint : null
+  thumbprintSecondary: !empty(reverseProxyCertificate) ? reverseProxyCertificate.thumbprintSecondary : null
+  x509StoreName: !empty(reverseProxyCertificate) ? reverseProxyCertificate.x509StoreName : null
+}
+
+var reverseProxyCertificateCommonNamesList_var = [for reverseProxyCertificateCommonName in items(reverseProxyCertificateCommonNames): {
```

Comment:
`items` should not be necessary if `reverseProxyCertificateCommonNames` is configured as an array

Reply:
However, it is not - which begs the question why it is an object, not an array?

Reply:
Yes, it wouldn't be, but `reverseProxyCertificateCommonNames` is an object in the original full spec template file, idea was to keep the data type unchanged and perform data manipulation inside the bicep script.